rust/src/builtins/scriptlet_intercept/groupadd.rs

```diff
@@ -0,0 +1,150 @@
+//! CLI handler for intercepted `groupadd`.
+
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::Result;
+use cap_std::fs::Dir;
+use clap::{Arg, Command};
+use std::io::Write;
+
+/// Entrypoint for (the rpm-ostree implementation of) `groupadd`.
+pub(crate) fn entrypoint(args: &[&str]) -> Result<()> {
+    fail::fail_point!("intercept_groupadd_ok", |_| Ok(()));
+
+    // This parses the same CLI surface as the real `groupadd`,
+    // but in the end we only extract the group name and (if
+    // present) the static GID.
+    let matches = cli_cmd().get_matches_from(args);
+    let gid: Option<u32> = match matches.value_of("gid") {
+        None => None,
+        Some(s) => Some(s.parse()?),
+    };
+    let groupname = matches
+        .value_of("groupname")
+        .expect("missing required groupname");
+
+    let rootdir = Dir::open_ambient_dir("/", cap_std::ambient_authority())?;
+    generate_sysusers_fragment(&rootdir, groupname, gid)?;
+
+    Ok(())
+}
+
+/// CLI parser, matches <https://linux.die.net/man/8/groupadd>.
+fn cli_cmd() -> Command<'static> {
```

Comment:
Hmm, shouldn't we error out if any of the switches we don't actually support are used? And I guess one way to do that is to just drop those switches from the clap definition.

Reply:
Likely yes. I'll do that in a followup PR together with enforcing the `--system` flag, as soon as I've verified that all packages in the FCOS base image are fine.